File: app/src/main/java/com/mapelli/simone/githubclient/UI/SplashScreenActivity.java
package com.mapelli.simone.githubclient.UI;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;
import android.os.Handler;

import com.mapelli.simone.githubclient.R;


/**
 * -------------------------------------------------------------------------------------------------
 * Minimal splash screen
 */
public class SplashScreenActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);
        // startAppDirect();
        startAppWithDelay();
        loadingHandler();

    }

    private void startAppDirect(){
        Intent intent = new Intent(getApplicationContext(), SearchUsersActivity.class);
        startActivity(intent);
        finish();
    }

    private void startAppWithDelay() {
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(getApplicationContext(), SearchUsersActivity.class);
                startActivity(intent);
                finish();
            }
        }, 1500);
    }


    // Loading indicator
    private void loadingHandler() {
        final TextView loadingTxt = findViewById(R.id.loading_splash_txt);
        final Handler handler = new Handler();
        final Runnable runnable = new Runnable() {
            @Override
            public void run() {
                if(loadingTxt.getText().length() > 10) {
                    loadingTxt.setText("Loading ");
                } else {
                    loadingTxt.setText(loadingTxt.getText()+".");
                }
                handler.postDelayed(this, 500);
            }
        };
        handler.postDelayed(runnable, 500);
    }
}

File: app/src/main/java/com/mapelli/simone/githubclient/UI/SplashScreenActivity.kt
package com.mapelli.simone.githubclient.UI

import androidx.appcompat.app.AppCompatActivity

import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import android.os.Handler

import com.mapelli.simone.githubclient.R


/**
 * -------------------------------------------------------------------------------------------------
 * Minimal splash screen
 */
class SplashScreenActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)
        startAppWithDelay()
        loadingHandler()

    }

    private fun startAppDirect() {
        val intent = Intent(applicationContext, UsersSearchActivity::class.java)
        startActivity(intent)
        finish()
    }

    private fun startAppWithDelay() {
        Handler().postDelayed({
            val intent = Intent(applicationContext, UsersSearchActivity::class.java)
            startActivity(intent)
            finish()
        }, 1500)
    }


    // Loading indicator
    private fun loadingHandler() {
        val loadingTxt = findViewById<TextView>(R.id.loading_splash_txt)
        val handler = Handler()
        val runnable = object : Runnable {
            override fun run() {
                if (loadingTxt.text.length > 10) {
                    loadingTxt.text = "Loading "
                } else {
                    loadingTxt.text = loadingTxt.text.toString() + "."
                }
                handler.postDelayed(this, 500)
            }
        }
        handler.postDelayed(runnable, 500)
    }
}
